minimal.js

```diff
@@ -226,38 +239,68 @@ module.exports = function (dirname, keys, opts) {
   }
 
   db.addUnboxer = function (unboxer) {
-    unboxers.push(unboxer)
+    switch (typeof unboxer) {
+      case 'function':
+        unboxers.push(unboxer)
+        break
+
+      case 'object':
+        if (typeof unboxer.key !== 'function') throw new Error('invalid unboxer')
+        if (typeof unboxer.value !== 'function') throw new Error('invalid unboxer')
+        unboxers.push(unboxer)
+        break
+
+      default: throw new Error('invalid unboxer')
+    }
   }
 
   /* TODO extract to ssb-private */
   var box1 = {
-    box: function (content, recps) {
-      if (!recps.every(isFeed)) return null
+    boxer: function (content, recps, cb) {
+      if (!recps.every(isFeed)) return cb(null, null)
 
-      return ssbKeys.box(content, recps)
+      cb(null, ssbKeys.box(content, recps))
     },
-    unbox: {
-      key: function (ciphertext) { return ssbKeys.unboxKey(ciphertext, keys) },
-      value: function (ciphertext, key) { return ssbKeys.unboxBody(ciphertext, key) }
+    unboxer: {
+      key: function (ciphertext, value, cb) { 
+        if (!ciphertext.endsWith('.box')) return cb(null, null)
+        // TODO move this inside of ssb-keys
+
+        cb(null, ssbKeys.unboxKey(ciphertext, keys))
+      },
+      value: function (ciphertext, msgKey, value, cb) {
+        cb(null, ssbKeys.unboxBody(ciphertext, msgKey))
+      }
```

Comment:
I'd like a review on the signatures of these boxers / unboxers.
What adding a `cb` arg to the end surfaced is that the current signature has args which aren't being used (i.e. the `value = msg.value`).

I am tempted to refactor these functions to have signatures like : 

```js
function unboxValue (ciphertext, msgKey, context, cb) {
  // ...
}
```
where `context = { msg, ...futureThings }`

I'd also change the `unboxer` signature to `unboxer = { unboxKey: Function, unboxValue: Function }`

Reply:
@christianbundy said : 

> I'm sure there's a good reason why, but I don't understand:  could this just be `unboxValue(msg, cb)`? Seems like we've destructured `msg` into its parts and are passing `content` twice (in `msg.value.content` and `msg.value`).

Reply:
reviewed and pruned with Dominic